### ISSUE TITLE
Borsa

### ISSUE THREAD

**Issue by ofedragon-lgtm** [streamlit/blank-app-template#68]
import streamlit as st
import yfinance as yf
import pandas as pd
import numpy as np
import plotly.graph_objects as go
from datetime import datetime, timedelta

# Mobil Sayfa Ayarları
st.set_page_config(page_title="Borsa Sinyal Mobil", layout="centered", page_icon="📱")

st.title("📱 Mobil Borsa Analiz")
st.caption("Tam ekran mobil uyumlu teknik analiz ve sinyal motoru.")

# Hisse Giriş Alanı (Büyük ve kolay dokunulabilir)
ticker = st.text_input("Hisse Kodu Girin (Örn: THYAO.IS, AAPL)", value="THYAO.IS").upper()

# Zaman aralığı seçimi (Mobil uyumlu butonlar)
period_choice = st.radio("Analiz Dönemi", ["6 Aylık", "1 Yıllık"], horizontal=True)
period_map = {"6 Aylık": 180, "1 Yıllık": 365}

start_date = (datetime.today() - timedelta(days=period_map[period_choice])).strftime('%Y-%m-%d')
end_date = datetime.today().strftime('%Y-%m-%d')

# Teknik Hesaplamalar
def calculate_indicators(df):
    if isinstance(df.columns, pd.MultiIndex):
        df.columns = df.columns.droplevel(1)
    df['SMA20'] = df['Close'].rolling(window=20).mean()
    df['SMA50'] = df['Close'].rolling(window=50).mean()
    
    delta = df['Close'].diff()
    gain = (delta.where(delta > 0, 0)).rolling(window=14).mean()
    loss = (-delta.where(delta < 0, 0)).rolling(window=14).mean()
    rs = gain / (loss + 1e-10)
    df['RSI'] = 100 - (100 / (1 + rs))
    
    exp1 = df['Close'].ewm(span=12, adjust=False).mean()
    exp2 = df['Close'].ewm(span=26, adjust=False).mean()
    df['MACD'] = exp1 - exp2
    df['MACD_Signal'] = df['MACD'].ewm(span=9, adjust=False).mean()
    return df

def generate_signal(df):
    latest = df.iloc[-1]
    prev = df.iloc[-2]
    score = 0
    reasons = []
    
    if latest['RSI'] < 30: score += 2; reasons.append("🟢 **RSI Al:** Hisse aşırı satıldı, tepki yükselişi gelebilir.")
    elif latest['RSI'] > 70: score -= 2; reasons.append("🔴 **RSI Sat:** Fiyat şişti, kar satışı gelebilir.")
    
    if latest['Close'] > latest['SMA20'] > latest['SMA50']: score += 2; reasons.append("🟢 **Trend Olumlu:** Yükseliş trendi güçlü şekilde sürüyor.")
    elif latest['Close'] < latest['SMA20'] < latest['SMA50']: score -= 2; reasons.append("🔴 **Trend Olumsuz:** Düşüş trendi devam ediyor.")
    
    if latest['MACD'] > latest['MACD_Signal'] and prev['MACD'] <= prev['MACD_Signal']: score += 1.5; reasons.append("🟢 **MACD Al Sinyali:** Pozitif momentum başlıyor.")
    elif latest['MACD'] < latest['MACD_Signal'] and prev['MACD'] >= prev['MACD_Signal']: score -= 1.5; reasons.append("🔴 **MACD Sat Sinyali:** Güç kaybı başladı.")
    
    if score >= 2: verdict, color = "GÜÇLÜ AL", "rgba(46, 204, 113, 0.2)"
    elif 0.5 <= score < 2: verdict, color = "KADEMELİ AL", "rgba(241, 196, 15, 0.2)"
    elif -0.5 < score < 0.5: verdict, color = "NÖTR / BEKLE", "rgba(149, 165, 166, 0.2)"
    elif -2 < score <= -0.5: verdict, color = "KADEMELİ SAT", "rgba(230, 126, 34, 0.2)"
    else: verdict, color = "GÜÇLÜ SAT", "rgba(231, 76, 60, 0.2)"
    
    return verdict, reasons, color

# Veriyi İşleme ve Ekrana Basma
try:
    data = yf.download(ticker, start=start_date, end=end_date)
    if not data.empty:
        df = calculate_indicators(data.copy())
        verdict, reasons, bg_color = generate_signal(df)
        
        latest_price = float(df['Close'].iloc[-1])
        prev_price = float(df['Close'].iloc[-2])
        change_pct = ((latest_price - prev_price) / prev_price) * 100
        
        # Mobil Bilgi Kartları (Dikey Düzen)
        st.metric("Son Fiyat", f"{latest_price:.2f}", delta=f"{change_pct:.2f}%")
        
        # Sinyal Kutusu
        st.markdown(f"""
        <div style="background-color:{bg_color}; padding:15px; border-radius:8px; text-align:center; margin:10px 0;">
            <h3 style="margin:0; color:black;">Sinyal: {verdict}</h3>
        </div>
        """, unsafe_allow_html=True)
        
        # Gerekçeler
        st.subheader("📝 Sinyal Detayları")
        for r in reasons:
            st.markdown(r)
            
        # Mobil Uyumlu Grafik (Dar alanlar için optimize edildi)
        st.subheader("📊 Fiyat Grafiği")
        fig = go.Figure()
        fig.add_trace(go.Scatter(x=df.index, y=df['Close'], name='Fiyat', line=dict(color='black', width=2)))
        fig.add_trace(go.Scatter(x=df.index, y=df['SMA20'], name='20 Günlük', line=dict(color='blue', width=1)))
        fig.update_layout(
            margin=dict(l=10, r=10, t=10, b=10),
            height=300,
            template="plotly_white",
            legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="right", x=1)
        )
        st.plotly_chart(fig, use_container_width=True)
    else:
        st.error("Veri bulunamadı. BIST için kodun sonuna '.IS' ekleyin.")
except Exception as e:
    st.error("Bağlantı veya sembol hatası.")
